helmfile/values/rclone-sync.yaml.gotmpl

```
@@ -11,10 +16,29 @@ config:
   dryrun: {{ .Values.objectStorage.sync.dryrun }}
 {{- end }}
 
+{{ $sourceS3 := "" }}
+{{ $sourceSwift := "" }}
+{{ $destinationS3 := "" }}
+{{ $destinationSwift := "" }}
+
+{{- range .Values.objectStorage.sync.buckets }}
+{{- if or (eq (get "sourceType" "" .) "s3") (and (ne (get "sourceType" "" .) "swift") (eq $.Values.objectStorage.type "s3") ) }}
+{{ $sourceS3 = "true" }}
+{{- else if or (eq (get "sourceType" "" .) "swift") (and (ne (get "sourceType" "" .) "s3") (eq $.Values.objectStorage.type "swift") ) }}
+{{ $sourceSwift = "true" }}
+{{- end }}
+
+{{- if or (eq (get "destinationType" "" .) "s3") (and (ne (get "destinationType" "" .) "swift") (eq $.Values.objectStorage.sync.type "s3") ) }}
+{{ $destinationS3 = "true" }}
+{{- else if or (eq (get "destinationType" "" .) "swift") (and (ne (get "destinationType" "" .) "s3") (eq $.Values.objectStorage.sync.type "swift") ) }}
+{{ $destinationSwift = "true" }}
+{{- end }}
+{{- end }}
+
   source:
-    name: default
-    type: {{ .Values.objectStorage.type }}
-  {{- if eq .Values.objectStorage.type "s3" }}
+  {{- if ($sourceS3) }}
```

Let's drop parenthesis around if statements with only one argument
```suggestion
  {{- if $sourceS3 }}
```